### ISSUE TITLE
docopt incorrectly parses options in argv when these are passed together, e.g. in a script

### ISSUE THREAD

**Issue by mintigo-efi** [docopt/docopt#268]
```
$ cat s.py 
from docopt import docopt
import sys

__doc__ = """ 
My cool script!

Usage:
  s -h | --help
  s one [options]
  s two [options]

Global options:
  --file=<path>          read file
  --verbose              turn on verbose output
  --sleep                go to sleep in the end 

""" 

print sys.argv

opts = docopt(__doc__)
print "Hey, now I can run!"
print opts

$ python s.py one --verbose --sleep
['s.py', 'one', '--verbose', '--sleep']
Hey, now I can run!
{'--file': None,
 '--help': False,
 '--sleep': True,
 '--verbose': True,
 '-h': False,
 'one': True,
 'two': False}

$ cat q.sh
#!/bin/bash
which=one
file="--file=path/to/file"
other="--verbose --sleep"

echo Running: python my-runner "$which" "$file" "$other"
python s.py "$which" "$file" "$other"

$ chmod 755 q.sh
$ ./q.sh
Running: python my-runner one --file=path/to/file --verbose --sleep
['s.py', 'one', '--file=path/to/file', '--verbose --sleep']
Usage:
  s -h | --help
  s one [options]
  s two [options]
```

Here, the last parameter ```--verbose --script``` is not parsed correctly. As a result, Usage is printed which is not very helpful, as copying the result of ```echo``` and running it would do just fine.

Also, please consider adding more verbatim output of what exactly went wrong with the parameters instead of just printing usage string.

**Comment by BlaXpirit** [comment]
[~]$ cat "--show-ends --show-tabs" /etc/passwd
cat: unrecognized option '--show-ends --show-tabs'
Try 'cat --help' for more information.

**Comment by keleshev** [comment]
Yes, this is how shells work overall. Here's a simplest example:

```sh
$ options="-l -h" 
$ ls "$options"  # expands into: ls "-l -h" (which doesn't work)
ls: illegal option --  
usage: ls [-ABCFGHLOPRSTUWabcdefghiklmnopqrstuwx1] [file ...]
$ ls $options  # expands into: ls -a -h (which works)
-rw-r--r--@    1 vladimir  staff    56K Sep 25  2014 README.txt
```

**Comment by keleshev** [comment]
To debug this kind of issue you can define `argv-of` like this:

$ alias argv-of='ruby -e "p ARGV"'
$ argv-of ls $options
["ls", "-a", "-h"]
$ argv-of ls "$options"
["ls", "-a -h"]
```

**Comment by mintigo-efi** [comment]
```
cat: unrecognized option '--show-ends --show-tabs'
```
Note how easy it is to understand the issue, and how much time it should take to come up with the solution.

**Comment by BlaXpirit** [comment]
So come up with your own workaround if you want nonstandard, error-prone behavior.

**Comment by mintigo-efi** [comment]
I now understand that this is standard behaviour, there is no need to personally attack me, Oleh. All I'm advocating for currently is a better error statement that would allow the person to understand where the parsing went wrong. Something to the tune of 

```
Unrecognized option: '--verbose --sleep'
Usage:
  s -h | --help
<...>
```

**Comment by mintigo-efi** [comment]
@keleshev it looks like I can't define alias in a script and use it. Gets me ```argv-of: command not found```.

**Comment by BurntSushi** [comment]
@mintigo-efi I'm in favor of better failure modes, although it can be difficult. The way that `argv` is matched against usage patterns makes it pretty tricky to describe *why* no match could be found. It's a more a binary "here's the usage that it matched or no match." I'd like to work on improving this some day. :)

**Comment by mintigo-efi** [comment]
Then describe *where* no match is found.

Here I am 12 hours later, running a job in Jenkins that configures another job with parameters that runs a shell script that runs Python script that uses docopt. Docopt parsing fails, and all I get from docopt is a dump of how to use my script, and I still have no idea why.

**Comment by BurntSushi** [comment]
> Then describe where no match is found.

That's what I just said was hard.

> Here I am 12 hours later, running a job in Jenkins that configures another job with parameters that runs a shell script that runs Python script that uses docopt. Docopt parsing fails, and all I get from docopt is a dump of how to use my script, and I still have no idea why.

Yes. I just agreed with you. Better failure modes would be great.

**Comment by keleshev** [comment]
@mintigo-efi regarding alias not working in script—you could use `ruby -e "p ARGV"` inline, or define a small Ruby/Python script that does that. Probably it's possible to write a bash function for that, but I tried that now and failed, so can't help here. In general I try to avoid writing shell scripts at all costs and use shell only for quick commands in the terminal, leaving any multi-line scripting to Ruby/Python.

Regarding docopt error-handling in general—I really want to improve it, I only wish I had more time.
